compositor_pipeline/src/pipeline.rs

```diff
@@ -112,7 +116,7 @@ impl Pipeline {
     pub fn register_input(
         &mut self,
         register_options: RegisterInputOptions,
-    ) -> Result<Port, RegisterInputError> {
+    ) -> Result<Option<Port>, RegisterInputError> {
```

Comment:
some enum would probably be a better fit, but with just 2 formats it's fine if you want to leave it

Reply:
yeah I left it like this with the intention to handle it in a separate PR, bc I have a feeling other refactoring opportunities will surface as I do other mp4-related stuff